<launch>
<param name="/use_sim_time" value="true"/>

  <!-- Play in quiet mode so roslaunch output stays readable -->
  <node pkg="rosbag"
        type="play"
        name="bag_player"
        output="screen"
        args="/home/salahuddin/cornell/IR/IR-VO/rosbags/kitti/kitti_00.bag --clock -q"/>

  <node pkg="ir_vo" type="stereo_vo_node.py" name="stereo_vo" output="screen">
    <!-- Remap left camera topics -->
    <remap from="/left_camera/image_raw"     to="/kitti_stereo/left/image_rect"/>
    <remap from="/left_camera/camera_info"   to="/kitti_stereo/left/camera_info"/>
    <!-- Remap right camera topics -->
    <remap from="/right_camera/image_raw"    to="/kitti_stereo/right/image_rect"/>
    <remap from="/right_camera/camera_info"  to="/kitti_stereo/right/camera_info"/>
    <param name="detector" value="orb"/>
  </node>

  <node pkg="rviz"
        type="rviz"
        name="rviz"
        required="true"
        args="-d $(find ir_vo)/viz/kitti.rviz"/>

</launch> 